core/src/main/kotlin/com/justai/jaicf/reactions/Reaction.kt

```diff
@@ -0,0 +1,85 @@
+package com.justai.jaicf.reactions
+
+import com.justai.jaicf.context.LoggingContext
+
+/**
+ * Abstraction for result of performing some reaction.
+ *
+ * @see Reactions
+ * */
+interface Reaction
```

Comment:
Every implementation has `state` property - a signal to make this abstract (also please look at the comment below regarding the obfuscation).